pkg/factory/factory.go

```diff
@@ -0,0 +1,39 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package factory
+
+import (
+	"fmt"
+	"github.com/edgexfoundry/go-mod-registry"
+	"github.com/edgexfoundry/go-mod-registry/internal/pkg/consul"
+)
+
+func NewRegistryClient(registryConfig registry.Config, serviceId string) (registry.Client, error) {
```

Comment:
What is `serviceId`? Is that `serviceKey`? If so, please rename for consistency. Could possibly be added to the registry.Config type.

Reply:
Yes and yes it could be added to the registry.Config type. Will that.